tools/compatibility/v2/ast_transformer.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""An ast.NodeTransformer subclass for updating SageMaker Python SDK code."""
+from __future__ import absolute_import
+
+import ast
+
+from modifiers import framework_version
+
+FUNCTION_CALL_MODIFIERS = [framework_version.FrameworkVersionEnforcer()]
+
+
+class ASTTransformer(ast.NodeTransformer):
+    """An ``ast.NodeTransformer`` subclass that walks the abstract syntax tree and
+    modifies nodes to upgrade the given SageMaker Python SDK code.
+    """
+
+    def visit_Call(self, node):
```

Comment:
Should this be snake case (ie. `visit_call`)? Or is this naming necessary to override a method in `NodeTransformer`?

Reply:
it's how `ast` names it, unfortunately. Based on https://docs.python.org/3/library/ast.html#ast.NodeVisitor.visit, I suspect there might be something programmatic about it, and since all the class names are capitalized, so are the function names 🤷‍♀️